ssl-config/src/main/java/com/palantir/remoting/ssl/SslConfiguration.java

```diff
@@ -20,15 +20,29 @@
 import com.fasterxml.jackson.databind.annotation.JsonSerialize;
 import com.google.common.base.Optional;
 import com.google.common.base.Preconditions;
+import com.google.common.base.Throwables;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.file.Files;
 import java.nio.file.Path;
+import java.security.GeneralSecurityException;
+import java.security.Key;
+import java.security.KeyStore;
+import java.security.cert.Certificate;
+import javax.net.ssl.KeyManagerFactory;
+import javax.net.ssl.TrustManagerFactory;
 import org.immutables.value.Value;
 import org.immutables.value.Value.Style.ImplementationVisibility;
 
+/**
+ * Configuration that provides a {@link TrustManagerFactory} based on a trust store and optionally provides
+ * a {@link KeyManagerFactory} based on a key store. The stores can be in JKS or PKCS12 format.
+ */
 @JsonDeserialize(as = ImmutableSslConfiguration.class)
 @JsonSerialize(as = ImmutableSslConfiguration.class)
 @Value.Immutable
 @Value.Style(visibility = ImplementationVisibility.PACKAGE)
-public abstract class SslConfiguration {
+public abstract class SslConfiguration implements SslConfigurationProvider {
```

Comment:
Yup, Puppet-style directory makes sense, and am fine with that.

The one other use case that I imagined/find myself using locally sometimes is wanting to point to a single PKCS#12 cert (in PEM or DER format) and using that as the trust source. To support this in the Puppet scenario, I would need to create `certs` directory inside the base directory and place the PEM file there.

This isn't too onerous, but still requires some movement/modification of files locally (versus being able to have a config that points to a specific file that's known to exist in a location).